### PR TITLE
fix specs disappearing on attacking/killing vorkath spawn and corp core

### LINKED ISSUE #4678: BGS and DWH spec counters go away at Vorkath when crumble undeading the crab
Anytime you are killing vorkath and you target the zombie crab with crumble undead, when you return to killing vorkath it no longer shows the spec amount you hit.

While this is not gamebreaking by any means lol, it is nice to be able to see what you specced during the entire fight

How to reproduce bug:
1.) Go to vorkath and hit a Bandos Godsword special attack on vorkath. 
2.) Wait until the crab spawns and cast crumble undead on the crab killing it.
3.) Return to killing vorkath and the special indicator is no longer available to be seen.

Why this happens:

I believe the Special Attack counter is written to disappear anytime a target is switched so when you go to kill the crab with the spell it counts the crab as a new target and forgets the special you hit on vorkath. 

Solution:

Possibly edit the section of code to remember what has been special attacked in the last 5 minutes or X amount of targets ago and any time that target is returned to display the counter again. 

-I can not code, so if my solution doesn't make sense I apologize, mad respect for you lads who can-

### PR DESCRIPTION
**Pull request by winterdaze**
#2741 (kq, sire phases) should already have been fixed in a previous update
Closes #4678

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/SpecialCounterPlugin.java [hunk 1]
```diff
@@ -66,6 +68,7 @@
 	private int specialPercentage;
 	private Actor lastSpecTarget;
 	private int lastSpecTick;
+	private final int[] npcWhitelist = {NpcID.ZOMBIFIED_SPAWN_8063, NpcID.ZOMBIFIED_SPAWN, NpcID.DARK_ENERGY_CORE};
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/SpecialCounterPlugin.java]: I'd rather this be a property of `Boss` so we don't have some special case in the plugin that's not apparent when reading the enum.

File: runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/Boss.java [hunk 2]
```diff
@@ -38,20 +38,33 @@
...
+	VETION(NpcID.VETION, NpcID.VETION_REBORN),
+	VORKATH(new Integer[]{NpcID.ZOMBIFIED_SPAWN, NpcID.ZOMBIFIED_SPAWN_8063}, NpcID.VORKATH, NpcID.VORKATH_8058, NpcID.VORKATH_8059, NpcID.VORKATH_8060, NpcID.VORKATH_8061);
 
 	private final Set<Integer> ids;
+	private final Set<Integer> minionIds;
 
 	Boss(Integer... ids)
 	{
 		this.ids = Sets.newHashSet(ids);
+		this.minionIds = Sets.newHashSet(new Integer[]{});
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/Boss.java]:
```suggestion
		this.minionIds = Collections.emptySet();
```

File: runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/Boss.java [hunk 3]
```diff
@@ -38,20 +38,33 @@
...
+	}
+
+	Boss(Integer[] minionIds, Integer... bossIds)
+	{
+		Set<Integer> npcIds = Sets.newHashSet(minionIds);
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/Boss.java]: Using this as the basis for the boss ID set means that hitting a minion with a special attack would generate a special attack infobox, which is not desirable.

File: runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/SpecialCounterPlugin.java [hunk 4]
```diff
@@ -238,7 +238,15 @@ public void onNpcDespawned(NpcDespawned npcDespawned)
 
 		if (actor.isDead() && interactedNpcIds.contains(actor.getId()))
 		{
-			removeCounters();
+			if (Boss.getBoss(actor.getId()) == null)
+			{
+				removeCounters();
+				return;
+			}
+			if (Boss.getBoss(actor.getId()).getMinionIds().size() == 0 || (Boss.getBoss(actor.getId()).getMinionIds().size() > 0 && !Boss.getBoss(actor.getId()).getMinionIds().contains(actor.getId())))
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/SpecialCounterPlugin.java:246]: ?

**Comment by Nightfirecat** [comment]
This was added in aa1cd2c7ff79e368db6769d1c09f6b8a11c917d9.

**Pull request closed without merging**